test/hotspot/jtreg/compiler/types/TestCheckCastPPBecomesTOP.java

```diff
@@ -25,6 +25,7 @@
  * @test
  * @bug 8297345
  * @summary C2: SIGSEGV in PhaseIdealLoop::push_pinned_nodes_thru_region
+ * @requires vm.gc.Parallel
```

Comment:
I think it should be:

```suggestion
 * @requires vm.gc == null | vm.gc.Parallel
```

Reply:
FTR, I discussed this privately with Tobias. His concern is that the test would not be run if not passed `-XX:+UseParallelGC`. That doesn't seem to be an issue though as the test runs as expected when passed no option or `-XX:+UseParallelGC` and doesn't run with a command line option that enables some other gc.

Reply:
Which means it will run with default GC: G1.

Reply:
Which is the correct way? `@requires vm.gc == null | vm.gc.Parallel` or `@requires vm.gc.Parallel`?

Reply:
`@requires vm.gc.Parallel` 

 [JDK-8160088](https://bugs.openjdk.org/browse/JDK-8160088)  changed it from original `@requires vm.gc == "null" | vm.gc == "Parallel"`

Reply:
Clarifying. `vm.gc.Parallel` is `true` when ParallelGC supported and its flag is specified on command line or no GC flags specified and GC is selected ergonomically. Which means the same as original `@requires`